app/code/Magento/Inventory/Model/SourceRepository.php

```diff
@@ -0,0 +1,127 @@
+ * See COPYING.txt for license details.
+ */
+
+namespace Magento\Inventory\Model;
+
+use Magento\Framework\Api\SearchCriteriaInterface;
+use Magento\Framework\Exception\CouldNotSaveException;
+use Magento\Framework\Api\SearchCriteria\CollectionProcessorInterface;
+use Magento\Framework\Exception\NoSuchEntityException;
+use Magento\Framework\Model\AbstractModel;
+use Magento\InventoryApi\Api\Data\SourceInterface;
+use Magento\InventoryApi\Api\Data\SourceSearchResultsInterface;
+use Magento\InventoryApi\Api\SourceRepositoryInterface;
+use Magento\Inventory\Model\SourceSearchResultsFactory;
+use Magento\Inventory\Model\Resource\Source as ResourceSource;
+use Magento\Inventory\Model\Resource\Source\CollectionFactory;
+use Magento\Inventory\Model\SourceFactory;
+
+/**
+ * Class SourceRepository
+ * @package Magento\Inventory\Model
+ */
+class SourceRepository implements SourceRepositoryInterface
+{
+    /**
+     * @var ResourceSource
+     */
+    private $resource;
+
+    /**
+     * @var SourceFactory
+     */
+    private $sourceFactory;
+
+    /**
+     * @var CollectionProcessorInterface
+     */
+    private $collectionProcessor;
+
+    /**
+     * @var CollectionFactory
+     */
+    private $collectionFactory;
+
+    /**
+     * @var SourceSearchResultsFactory
+     */
+    private $sourceSearchResultsFactory;
+
+    /**
+     * SourceRepository constructor.
+     * @param ResourceSource $resource
+     * @param SourceFactory $sourceFactory
+     * @param CollectionProcessorInterface $collectionProcessor
+     * @param CollectionFactory $collectionFactory
+     * @param SourceSearchResultsFactory $sourceSearchResultsFactory
+     */
+    public function __construct(
+        ResourceSource $resource,
+        SourceFactory $sourceFactory,
+        CollectionProcessorInterface $collectionProcessor,
+        CollectionFactory $collectionFactory,
+        SourceSearchResultsFactory $sourceSearchResultsFactory
+    ) {
+        $this->resource = $resource;
+        $this->sourceFactory = $sourceFactory;
+        $this->collectionProcessor = $collectionProcessor;
+        $this->collectionFactory = $collectionFactory;
+        $this->sourceSearchResultsFactory = $sourceSearchResultsFactory;
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function save(SourceInterface $source)
+    {
+        try {
+            $this->resource->save($source);
+        } catch (\Exception $exception) {
+            throw new CouldNotSaveException(__($exception->getMessage()));
+        }
+        return $source;
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function get($sourceId)
+    {
+        /** @var SourceInterface|AbstractModel $model */
+        $model = $this->sourceFactory->create();
+        $this->resource->load($model, SourceInterface::SOURCE_ID, $sourceId);
+
+        if (!$model->getSourceId()) {
+            throw NoSuchEntityException::singleField(SourceInterface::SOURCE_ID, $sourceId);
```

Comment:
` public static function singleField($fieldName, $fieldValue)
    {
        return new self(
            new Phrase(
                'No such entity with %fieldName = %fieldValue',
                [
                    'fieldName' => $fieldName,
                    'fieldValue' => $fieldValue
                ]
            )
        );
    }`
Can you look at this one? I think the static call makes sense, but i can change it if you want.